app/uk/gov/hmrc/transitmovements/controllers/MovementsController.scala

```diff
@@ -61,11 +62,11 @@ class MovementsController @Inject() (
             messageType <- extract(request.headers).asPresentation
             messageData <- xmlParsingService.extractMessageData(source, messageType).asPresentation
             fileSource = FileIO.fromPath(temporaryFile)
-            message <- factory.create(messageType, messageData.generationDate, Some(triggerId), fileSource).asPresentation
-            result  <- repo.updateMessages(DepartureId(movementId.value), message, messageData.mrn).asPresentation
-          } yield result).fold[Result](
+            message   <- factory.create(messageType, messageData.generationDate, Some(triggerId), fileSource).asPresentation
+            messageId <- repo.updateMessages(DepartureId(movementId.value), message, messageData.mrn).asPresentation
```

Comment:
As the ide isn't generated by the repository, but is generated by the message factory, this doesn't need to retrieve it. The block and just yield message.messageId